agent/consul/server.go

```diff
@@ -1056,21 +1056,13 @@ func (s *Server) setupRaft() error {
 			return nil
 		}
 		// Only use WAL if there is no existing raft.db, even if it's enabled.
-		if s.config.LogStoreConfig.Backend == LogStoreBackendDefault && !boltFileExists {
+		if (s.config.LogStoreConfig.Backend == LogStoreBackendDefault || s.config.LogStoreConfig.Backend == LogStoreBackendWAL) && !boltFileExists {
 			s.config.LogStoreConfig.Backend = LogStoreBackendWAL
```

Comment:
~ Should we consider moving the original
```go
if s.config.LogStoreConfig.Backend == LogStoreBackendDefault && !boltFileExists {
  s.config.LogStoreConfig.Backend = LogStoreBackendWAL
}
```
bit up above the rest of this `if` block, and just check explicitly for WAL (not default) after?

Main thought that crossed my mind is we're treating default and WAL as equivalent in these checks, so normalizing in one place is less error-prone and separates the defaulting from the business logic.

Reply:
Is that what you had in mind?

Reply:
I was more thinking we could simplify the checks following the first defaulting block, so that we aren't repeating stuff like "default or WAL" and "!boltDB". Maybe something like [this](https://github.com/hashicorp/consul/commit/4c4837cf96f877d2256babab6b93fd5838e24a6a?diff=split&w=0)? (also switches the warning to "using BoltDB" since "ignoring 'wal'" might be confusing when default is used)